process_test.go

```diff
@@ -0,0 +1,266 @@
+			},
+		),
+		Entry(
+			"when the handler configures an invalid key",
+			`*fixtures.ProcessMessageHandler is configured with an invalid identity, invalid key "\t \n", keys must be non-empty, printable UTF-8 strings with no whitespace`,
+			func(c dogma.ProcessConfigurer) {
+				c.Identity("<name>", "\t \n")
+				c.ConsumesEventType(fixtures.MessageA{})
+				c.ProducesCommandType(fixtures.MessageC{})
+			},
+		),
+		Entry(
+			"when the handler does not configure any consumed event types",
+			`*fixtures.ProcessMessageHandler is not configured to consume any events, ConsumesEventType() must be called at least once within Configure()`,
+			func(c dogma.ProcessConfigurer) {
+				c.Identity("<name>", "<key>")
+				c.ProducesCommandType(fixtures.MessageC{})
+			},
+		),
+		Entry(
+			"when the handler configures the same consumed event type multiple times",
+			`*fixtures.ProcessMessageHandler is configured to consume the fixtures.MessageA event more than once, should this refer to different message types?`,
+			func(c dogma.ProcessConfigurer) {
+				c.Identity("<name>", "<key>")
+				c.ConsumesEventType(fixtures.MessageA{})
+				c.ConsumesEventType(fixtures.MessageA{})
+				c.ProducesCommandType(fixtures.MessageC{})
+			},
+		),
+		Entry(
+			"when the handler does not configure any produced commands",
+			`*fixtures.ProcessMessageHandler is not configured to produce any commands, ProducesCommandType() must be called at least once within Configure()`,
+			func(c dogma.ProcessConfigurer) {
+				c.Identity("<name>", "<key>")
+				c.ConsumesEventType(fixtures.MessageA{})
+			},
+		),
+		Entry(
+			"when the handler configures the same produced command type multiple times",
+			`*fixtures.ProcessMessageHandler is configured to produce the fixtures.MessageC command more than once, should this refer to different message types?`,
+			func(c dogma.ProcessConfigurer) {
+				c.Identity("<name>", "<key>")
+				c.ConsumesEventType(fixtures.MessageA{})
+				c.ProducesCommandType(fixtures.MessageC{})
+				c.ProducesCommandType(fixtures.MessageC{})
+			},
+		),
+		Entry(
+			"when the handler configures the same scheudled timeout type multiple times",
+			`*fixtures.ProcessMessageHandler is configured to schedule the fixtures.MessageT timeout more than once, should this refer to different message types?`,
+			func(c dogma.ProcessConfigurer) {
+				c.Identity("<name>", "<key>")
+				c.ConsumesEventType(fixtures.MessageA{})
+				c.ProducesCommandType(fixtures.MessageC{})
+				c.SchedulesTimeoutType(fixtures.MessageT{})
+				c.SchedulesTimeoutType(fixtures.MessageT{})
+			},
+		),
+		Entry(
+			"when the handler configures the same message type with different roles",
+			`*fixtures.ProcessMessageHandler is configured to use fixtures.MessageA as both an event and a timeout`,
+			func(c dogma.ProcessConfigurer) {
+				c.Identity("<name>", "<key>")
+				c.ConsumesEventType(fixtures.MessageA{})
+				c.SchedulesTimeoutType(fixtures.MessageA{})
+			},
+		),
+	)
```

Comment:
Does this need to test any consuming of timeout types? When they are due?

Reply:
There's no actual messages in play, it's just the configuration stating what kinds of timeout messages are able to be scheduled. Because timeout messages are always sent back to the process instance that creates them, they are always added to both the produced and consumed lists.